src/code-templates/services/order-service/entry-points/api/server.ts

```diff
@@ -60,20 +61,24 @@ async function openConnection(
 function handleRouteErrors(expressApp: express.Application) {
   expressApp.use(
     async (
-      error: any,
+      error: unknown,
       req: express.Request,
       res: express.Response,
+      // Express requires next function in default error handlers
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
       next: express.NextFunction
     ) => {
-      if (typeof error === 'object') {
-        if (error.isTrusted === undefined || error.isTrusted === null) {
+      if (error instanceof AppError) {
+        if (error?.isTrusted === undefined || error.isTrusted === null) {
```

Comment:
⚠️ If its AppError, no need to do anything as the thrower already explicitly set this property

I think we should leave the code as-is

Reply:
Correct.
The current behavior is buggy too because `null` is `object` too.

From types POV this is a tough nut, we can't we export `normalizeError` function and then:
```typescript
function handleRouteErrors(expressApp: express.Application) {
  expressApp.use(
    async (
      error: unknown,
      req: express.Request,
      res: express.Response,
      // Express requires next function in default error handlers
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      next: express.NextFunction
    ) => {
      res.status(500);
      const normalizedError = normalizeError(error)
        if (Number.isInteger(error.HTTPStatus)) {
          // We need the `as` because of https://github.com/microsoft/TypeScript/issues/15048
          res.status(error.HTTPStatus as number);
        }
      // ✅ Best Practice: Pass all error to a centralized error handler so they get treated equally
      errorHandler.handleError(normalizedError);

      res.end();
    }
  );
}
```